test/test_linalg.py

```diff
@@ -2637,6 +2637,20 @@ def test_cholesky_solve_out_errors_and_warnings(self, device, dtype):
             self.assertEqual(len(w), 1)
             self.assertTrue("An output with one or more elements was resized" in str(w[-1].message))
 
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.double)
+    def test_cholesky_solve_backward(self, device, dtype):
+        b_dims = (5, 2)
+        L_dims = (5, 5)
+        upper = False
+
+        b = torch.randn(*b_dims, dtype=dtype, device=device, requires_grad=True)
+        L = torch.randn(*L_dims, dtype=dtype, device=device)
+        torch.autograd.gradcheck(lambda b: torch.cholesky_solve(b, L, upper=upper), (b,))
+        # TODO: Also check the gradient with respect to L. This seems to be messy with
+        # gradcheck because gradcheck isn't aware of the implicit symmetry in L.
```

Comment:
For this to work you would need to create an aux function that takes in unconstrained matrices, preprocesses them so that they are valid inputs for `cholesky_solve` (i.e. it calls `tril() or `triu()`) and then calls `cholesky_solve`.

Reply:
An extra `tril()` seems to fix the issue, thanks!